### PR TITLE
Tag helpers and validation

### PR DESCRIPTION
**Pull request by madeviv**
- Added tag helpers to simplify markup and error message logic
- Updated validation logic in date input model binder
- Added error service
- Changed redirect to use temp data instead of posting redirect value

### CODE REVIEW

File: ApplyToBecomeInternal/ApplyToBecomeInternal/Models/DateInputModelBinder.cs [hunk 1]
```diff
@@ -58,10 +68,30 @@ public Task BindModelAsync(ModelBindingContext bindingContext)
 				bindingContext.ModelState.SetModelValue(monthModelName, monthValueProviderResult);
 				bindingContext.ModelState.SetModelValue(yearModelName, yearValueProviderResult);
 
-				bindingContext.ModelState.TryAddModelError(
-					bindingContext.ModelName,
-					new FormatException("Invalid date specified."),
-					bindingContext.ModelMetadata);
+				if (_missingParts.Count == 3)
```
Comment (rebeccafitzsimmons1) [inline ApplyToBecomeInternal/ApplyToBecomeInternal/Models/DateInputModelBinder.cs:71]: I might be missing a requirement here but is it not a valid usecase that the user might want to enter the date later but update other fields now e.g. on the local authority page they could want to add comments and link but do the dates later?

File: ApplyToBecomeInternal/ApplyToBecomeInternal/ViewModels/SummaryListRowViewModel.cs [hunk 2]
```diff
@@ -2,7 +2,7 @@
 {
 	public class SummaryListRowViewModel
 	{
-		public SummaryListRowViewModel(string id, string key, string value, string page = null, string routeId = null, string hiddenText = null, bool altLayout = false)
+		/*public SummaryListRowViewModel(string id, string key, string value, string page = null, string routeId = null, string hiddenText = null, bool altLayout = false)
```
Comment (rebeccafitzsimmons1) [inline ApplyToBecomeInternal/ApplyToBecomeInternal/ViewModels/SummaryListRowViewModel.cs:5]: is this still needed?

**Review by rebeccafitzsimmons1 — APPROVED** [review]
Looks really good, might be worth adding some tests around the validation

File: ApplyToBecomeInternal/ApplyToBecomeInternal/Models/DateInputModelBinder.cs (hunk 1 above)
Comment (madeviv) [reply ApplyToBecomeInternal/ApplyToBecomeInternal/Models/DateInputModelBinder.cs:69]: Ok its optional now

**Pull request merged**